### ISSUE TITLE
cmd/compile: bug in cmd/compile/internal/bitvec.NewBulk

### ISSUE THREAD

**Issue by prattmic** [golang/go#71003]
Issue created by [stacks](https://pkg.go.dev/golang.org/x/tools/gopls/internal/telemetry/cmd/stacks).

This stack `Ax9OyA` was [reported by telemetry](https://storage.googleapis.com/prod-telemetry-merged/2024-12-24.json):

- `compile/bug`
- [`cmd/compile/internal/base.FatalfAt:+3`](https://cs.opensource.google/go/go/+/go1.23.3:src/cmd/compile/internal/base/print.go;l=215)
- `cmd/compile/internal/base.Fatalf:=195`
- [`cmd/compile/internal/bitvec.NewBulk:+4`](https://cs.opensource.google/go/go/+/go1.23.3:src/cmd/compile/internal/bitvec/bv.go;l=40)
- [`cmd/compile/internal/liveness.newliveness:=434`](https://cs.opensource.google/go/go/+/go1.23.3:src/cmd/compile/internal/liveness/plive.go;l=434)
- [`cmd/compile/internal/liveness.Compute:+3`](https://cs.opensource.google/go/go/+/go1.23.3:src/cmd/compile/internal/liveness/plive.go;l=1380)
- [`cmd/compile/internal/ssagen.genssa:+6`](https://cs.opensource.google/go/go/+/go1.23.3:src/cmd/compile/internal/ssagen/ssa.go;l=7285)
- [`cmd/compile/internal/ssagen.Compile:+11`](https://cs.opensource.google/go/go/+/go1.23.3:src/cmd/compile/internal/ssagen/pgen.go;l=312)
- [`cmd/compile/internal/gc.compileFunctions.func5.1:+1`](https://cs.opensource.google/go/go/+/go1.23.3:src/cmd/compile/internal/gc/compile.go;l=188)
- [`cmd/compile/internal/gc.compileFunctions.func2:+1`](https://cs.opensource.google/go/go/+/go1.23.3:src/cmd/compile/internal/gc/compile.go;l=142)
- [`cmd/compile/internal/gc.compileFunctions.func5:+4`](https://cs.opensource.google/go/go/+/go1.23.3:src/cmd/compile/internal/gc/compile.go;l=187)
- [`cmd/compile/internal/gc.compileFunctions:+77`](https://cs.opensource.google/go/go/+/go1.23.3:src/cmd/compile/internal/gc/compile.go;l=198)
- [`cmd/compile/internal/gc.Main:+246`](https://cs.opensource.google/go/go/+/go1.23.3:src/cmd/compile/internal/gc/main.go;l=306)
- [`main.main:+12`](https://cs.opensource.google/go/go/+/go1.23.3:src/cmd/compile/main.go;l=57)
- [`runtime.main:+125`](https://cs.opensource.google/go/go/+/go1.23.3:src/runtime/proc.go;l=272)
- `runtime.goexit:+0`
```
cmd/compile@go1.23.3 go1.23.3 darwin/arm64  (2)
```

**Comment by prattmic** [comment]
This stack `tl7JoA` was [reported by telemetry](https://telemetry.go.dev/data/2025-03-20):

- `compile/bug`
- [`cmd/compile/internal/base.FatalfAt:+3`](https://cs.opensource.google/go/go/+/go1.24.3:src/cmd/compile/internal/base/print.go;l=215)
- [`cmd/compile/internal/bitvec.NewBulk:+4`](https://cs.opensource.google/go/go/+/go1.24.3:src/cmd/compile/internal/bitvec/bv.go;l=41)
- [`cmd/compile/internal/liveness.newliveness:=430`](https://cs.opensource.google/go/go/+/go1.24.3:src/cmd/compile/internal/liveness/plive.go;l=430)
- [`cmd/compile/internal/liveness.Compute:+3`](https://cs.opensource.google/go/go/+/go1.24.3:src/cmd/compile/internal/liveness/plive.go;l=1376)
- [`cmd/compile/internal/ssagen.genssa:+6`](https://cs.opensource.google/go/go/+/go1.24.3:src/cmd/compile/internal/ssagen/ssa.go;l=6341)
- [`cmd/compile/internal/ssagen.Compile:+11`](https://cs.opensource.google/go/go/+/go1.24.3:src/cmd/compile/internal/ssagen/pgen.go;l=314)
- [`cmd/compile/internal/gc.compileFunctions.func5.1:+1`](https://cs.opensource.google/go/go/+/go1.24.3:src/cmd/compile/internal/gc/compile.go;l=189)
- [`cmd/compile/internal/gc.compileFunctions.func3.1:+1`](https://cs.opensource.google/go/go/+/go1.24.3:src/cmd/compile/internal/gc/compile.go;l=171)
- `runtime.goexit:+0`
```
cmd/compile@go1.24.3 go1.24.3 darwin/arm64 (32)
cmd/compile@go1.24.0 go1.24.0 darwin/arm64 (24)

**Comment by prattmic** [comment]
This stack `zdyCMA` was [reported by telemetry](https://telemetry.go.dev/data/2025-08-02):

- `compile/bug`
- [`cmd/compile/internal/base.FatalfAt:+3`](https://cs.opensource.google/go/go/+/go1.24.2:src/cmd/compile/internal/base/print.go;l=215)
- [`cmd/compile/internal/bitvec.NewBulk:+4`](https://cs.opensource.google/go/go/+/go1.24.2:src/cmd/compile/internal/bitvec/bv.go;l=41)
- [`cmd/compile/internal/liveness.newliveness:=430`](https://cs.opensource.google/go/go/+/go1.24.2:src/cmd/compile/internal/liveness/plive.go;l=430)
- [`cmd/compile/internal/liveness.Compute:+3`](https://cs.opensource.google/go/go/+/go1.24.2:src/cmd/compile/internal/liveness/plive.go;l=1376)
- [`cmd/compile/internal/ssagen.genssa:+6`](https://cs.opensource.google/go/go/+/go1.24.2:src/cmd/compile/internal/ssagen/ssa.go;l=6341)
- [`cmd/compile/internal/ssagen.Compile:+11`](https://cs.opensource.google/go/go/+/go1.24.2:src/cmd/compile/internal/ssagen/pgen.go;l=314)
- [`cmd/compile/internal/gc.compileFunctions.func5.1:+1`](https://cs.opensource.google/go/go/+/go1.24.2:src/cmd/compile/internal/gc/compile.go;l=189)
- [`cmd/compile/internal/gc.compileFunctions.func2:+1`](https://cs.opensource.google/go/go/+/go1.24.2:src/cmd/compile/internal/gc/compile.go;l=143)
- [`cmd/compile/internal/gc.compileFunctions.func5:+4`](https://cs.opensource.google/go/go/+/go1.24.2:src/cmd/compile/internal/gc/compile.go;l=188)
- [`cmd/compile/internal/gc.compileFunctions:+77`](https://cs.opensource.google/go/go/+/go1.24.2:src/cmd/compile/internal/gc/compile.go;l=199)
- [`cmd/compile/internal/gc.Main:+255`](https://cs.opensource.google/go/go/+/go1.24.2:src/cmd/compile/internal/gc/main.go;l=316)
- [`main.main:+12`](https://cs.opensource.google/go/go/+/go1.24.2:src/cmd/compile/main.go;l=57)
- [`runtime.main:+136`](https://cs.opensource.google/go/go/+/go1.24.2:src/runtime/proc.go;l=283)
- `runtime.goexit:+0`
```
cmd/compile@go1.24.2 go1.24.2 linux/amd64 (3)

**Comment by prattmic** [comment]
cc @golang/compiler 

This is by far the most commonly reported crash via telemetry. It sounds to me like a side-effect of a function that is too large?

**Comment by randall77** [comment]
Yes, this is often the first "your function is too big" ICE that people hit.
We should probably provide a better error message.

It's a late-in-compilation error so it isn't easy, but we do make a reasonable error message for, e.g., stack frame too big.
